src/utils/PropertyManager.C

```diff
@@ -14,8 +13,6 @@ declareCoefficient(PropertyMap<T, Tpw> & map,
                    std::shared_ptr<T> coef,
                    const ObjectManager<T> & manager)
 {
-  mooseAssert(std::find(manager.cbegin(), manager.cend(), coef) != manager.cend(),
-              "Coefficient object was not created by the appropriate coefficient manager.");
```

Comment:
I don't see a good reason to get rid of this as long as it remains possible for users to pass in arbitrary Coefficient objects, which could have been created anywhere.